replay-history/src/workflows.ts

```diff
@@ -0,0 +1,31 @@
+import * as wf from '@temporalio/workflow';
+
+export const getValueQuery = wf.defineQuery<number>('getValue');
+export const signals = {
+  add: wf.defineSignal<[number]>('add'),
+  sub: wf.defineSignal<[number]>('sub'),
+  div: wf.defineSignal<[number]>('div'),
+  mul: wf.defineSignal<[number]>('mul'),
+  inverseFraction: wf.defineSignal<[]>('inverseFraction'),
+};
+
+export async function calculator(): Promise<void> {
+  let value = 0;
+  wf.setHandler(signals.add, (x: number) => {
```

Comment:
I'd strongly suggest using a workflow that emits at least one command - users will want to experiment with replay failure, but this workflow is actually very very hard to make fail replay because added commands will not conflict with anything and the workflow never terminates. See https://www.notion.so/temporalio/A-fascinating-digression-on-workflow-compatibility-and-determinism-82c0582bf06e4152a6b08dc8d8f67463

Reply:
Roey's goal for this demo evolved towards just demonstrating step through debugging rather than demonstrating failure and then fixing it, but ofc i welcome suggestions